### PR TITLE
Fix bugs and provide flexibility for Softmax argument

### PR DESCRIPTION
**Pull request by dilyabareeva**
### Description
- The base metric class has softmax argument, but it is applied only for the predict method of model classes. quantus.explain function does not adjust softmax at all (issue #186).
- Softmax argument can now be passed to quantus.explain function and added to explain_func_kwargs in the metric call.
- Explanation function requires some important kwargs to be passed, like "method" and "xai_lib", but they are completely missing from the documentation. Added documentation for kwargs.
- generate_tf_explanation now also supports reduce_axes argument, in analogy to generate_captum_explanations. Sums up the output array along the given axes.
- Testing for 3-channel inputs for tf-explain added.

### Implemented changes
  - [x] Extend model interface with a method to extract model with softmax activation removed or added according to passed softmax argument.
  - [x] Create respective methods for PyTorch and Tensorflow.
  - [x] Fixed bug in pytorch_model: in the previous implementation, there was no check for an existing softmax layer, as if it was assumed that only model with no softmax in last layer are passed. 
  - [x] Add "softmax" and "reduce_axes" handling for the explanation function.
  - [x] Add user warnings about the appropriate use of the softmax argument.
  - [x] Add CIFAR10 model for tensorflow to tests.
  - [x] Added tests to the explanation function. 
  - [x] Documented the explanation function kwargs.

### CODE REVIEW

**Review by annahedstroem — COMMENTED** [review]
Super, no more comments, thank you.
